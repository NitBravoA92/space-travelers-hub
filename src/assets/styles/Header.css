header {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1;
  background-color: #fff;
  padding: 0.5rem 1rem 0.7rem;
  box-shadow: rgba(0, 0, 0, 0.1) 0 4px 6px -1px, rgba(0, 0, 0, 0.06) 0 2px 4px -1px;
}

.navbar {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 0.2rem;
}

.logo {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 0.8rem;
}

.logo-image {
  width: 100%;
  max-width: 45px;
}

.app-name {
  font-size: 1.5rem;
  font-weight: 600;
  color: #e59011;
}

#navigation {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 1rem;
}

.nav-item a {
  font-size: 1.3rem;
  font-weight: 500;
  color: #007bff;
  background-color: transparent;
}

.nav-item a.active,
.nav-item a:hover {
  text-decoration: underline;
}
