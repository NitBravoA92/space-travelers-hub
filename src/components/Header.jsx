import AppLogo from './AppLogo';
import MenuLink from './MenuLink';
import '../assets/styles/Header.css';

const Header = () => (
  <header>
    <div className="container">
      <nav className="navbar">
        <AppLogo />
        <ul id="navigation">
          <MenuLink url="/rockets" pageName="Rockets" />
          <MenuLink url="/missions" pageName="Missions" />
          <MenuLink url="/my-profile" pageName="My Profile" />
        </ul>
      </nav>
    </div>
  </header>
);

export default Header;
