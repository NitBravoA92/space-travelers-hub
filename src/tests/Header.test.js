import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Header from '../components/Header';

describe('The Header component', () => {
  test('renders into the DOM correctly', () => {
    const headerComponent = renderer
      .create(
        <MemoryRouter>
          <Header />
        </MemoryRouter>,
      )
      .toJSON();
    expect(headerComponent).toMatchSnapshot();
  });

  test("should render a '<nav>' element for the navigation menu with 3 link (<a>) elements", () => {
    const component = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );

    const navElement = component.container.querySelector('nav');
    const linksElements = component.container.querySelectorAll('.nav-item a');

    expect(navElement).toBeInTheDocument();
    expect(linksElements.length).toBe(3);
  });

  test("should render an '<img>' element with the class 'logo-image'", () => {
    const component = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );

    const imgElement = component.container.querySelector('.logo-image');

    expect(imgElement).toBeInTheDocument();
  });
});
