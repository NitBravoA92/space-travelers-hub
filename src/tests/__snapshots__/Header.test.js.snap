// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component renders into the DOM correctly 1`] = `
<header>
  <div
    className="container"
  >
    <nav
      className="navbar"
    >
      <a
        className="logo"
        href="/"
      >
        <img
          alt="Space Travelers' Hub"
          className="logo-image"
          src="logo.png"
        />
        <span
          className="app-name"
        >
          Space Travelers' Hub
        </span>
      </a>
      <ul
        id="navigation"
      >
        <li
          className="nav-item"
        >
          <a
            className={null}
            href="/rockets"
            onClick={[Function]}
          >
            Rockets
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className={null}
            href="/missions"
            onClick={[Function]}
          >
            Missions
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className={null}
            href="/my-profile"
            onClick={[Function]}
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
